flink/v1.16/flink/src/main/java/org/apache/iceberg/flink/sink/FlinkManifestUtil.java

```diff
@@ -73,21 +78,43 @@ static ManifestOutputFileFactory createOutputFileFactory(
         attemptNumber);
   }
 
+  /**
+   * Write the {@link WriteResult} to temporary manifest files.
+   *
+   * @param result all those DataFiles/DeleteFiles in this WriteResult should be written with same
+   *     partition spec
+   */
   static DeltaManifests writeCompletedFiles(
-      WriteResult result, Supplier<OutputFile> outputFileSupplier, PartitionSpec spec)
+      WriteResult result,
+      Supplier<OutputFile> outputFileSupplier,
+      Map<Integer, PartitionSpec> specsById)
       throws IOException {
 
     ManifestFile dataManifest = null;
     ManifestFile deleteManifest = null;
+    PartitionSpec spec = null;
 
     // Write the completed data files into a newly created data manifest file.
     if (result.dataFiles() != null && result.dataFiles().length > 0) {
+      int specId = result.dataFiles()[0].specId();
+      spec = specsById.get(specId);
+      Preconditions.checkState(
+          Arrays.stream(result.dataFiles()).allMatch(file -> file.specId() == specId),
```

Comment:
use `Arrays.stream(result.deleteFiles()).map(ContentFile::specId).distinct().count() > 1` ?

Reply:
Just curious, why do you prefer this? Is there any performance improvement?

Reply:
I just think this is more concise and easier to understand, we don’t need to take the first one out separately.